pydantic_ai_slim/pydantic_ai/mcp.py

```diff
@@ -363,17 +374,34 @@ async def client_streams(
         self,
     ) -> AsyncIterator[
         tuple[
-            MemoryObjectReceiveStream[JSONRPCMessage | Exception],
-            MemoryObjectSendStream[JSONRPCMessage],
+            MemoryObjectReceiveStream[SessionMessage | Exception],
+            MemoryObjectSendStream[SessionMessage],
         ]
     ]:  # pragma: no cover
-        async with sse_client(
+        if self.http_client and self.headers:
+            raise ValueError(f'In {self.__name__}, only one of `headers` or `http_client` can be provided.')
+
+        sse_client_partial = functools.partial(
+            sse_client,
             url=self.url,
-            headers=self.headers,
             timeout=self.timeout,
             sse_read_timeout=self.sse_read_timeout,
-        ) as (read_stream, write_stream):
-            yield read_stream, write_stream
+        )
+
+        if self.http_client is not None:
+
+            def httpx_client_factory(
+                headers: dict[str, str] | None = None,
+                timeout: httpx.Timeout | None = None,
+                auth: httpx.Auth | None = None,
+            ) -> httpx.AsyncClient:
+                return self.http_client  #  pyright: ignore
+
+            async with sse_client_partial(httpx_client_factory=httpx_client_factory) as (read_stream, write_stream):
+                yield read_stream, write_stream
+        else:
+            async with sse_client_partial(headers=self.headers) as (read_stream, write_stream):
+                yield read_stream, write_stream
```

Comment:
I don't think this is what we want. We either want:

1. Only allow `http_client` or `headers`/`timeout`/`auth`. It would be nice to add an `typing.overload`.
2. Allow all parameters, and replace them on the factory we are creating.